src/regdesk/states/reg_machine_manager.js

```diff
@@ -0,0 +1,106 @@
+import { LoadingState } from './loading_state.js';
+import { NewSearchState } from './new_search_state.js';
+import { SingleResultState } from './single_result_state.js';
+
+/**
+ * State mchine manager for the reg lookup interface.
+ */
+export class RegMachineManager {
+  /**
+   * Stores the state machine.
+   */
+  #machine;
+
+  /**
+   * Gets the state machine for this machine
+   */
+  get machine() {
+    return this.#machine;
+  }
+
+  /**
+   * Stores the current state of the machine.
+   */
+  #currentState;
+
+  /**
+   * Initializes a new instance of the RegMachineManager class.
+   * @param {RegMachineArgs} regMachineArgs - The standard record for each state class.
+   */
+  constructor(regMachineArgs) {
+    const machine = {
+      initial: NewSearchState.name,
+      states: {},
+    };
+
+    // TODO: Is there a better way to store this?
+    const eventMap = [
+      {
+        type: LoadingState,
+        events: {
+          [LoadingState.events.CANCEL]: NewSearchState.name,
+          [LoadingState.events.SINGLE_RESULT_READY]: SingleResultState.name,
+        },
+      },
+      {
+        type: NewSearchState,
+        events: {
+          [NewSearchState.events.START_SEARCH]: LoadingState.name,
+          [NewSearchState.events.RESET]: NewSearchState.name,
+        },
+      },
+      {
+        type: SingleResultState,
+        events: {
+          [SingleResultState.events.CANCEL]: NewSearchState.name,
+        },
+      },
+    ];
+
+    // TODO: Probably a way to merge the event map with the machine declaration
+    // whem I'm not mentally fried on my third flight of the day.
+    eventMap.forEach((state) => {
+      // Interesting, eslint doesn't handle this correctly.
+      // eslint-disable-next-line new-cap
+      machine.states[state.type.name] = new state.type(regMachineArgs, state.events);
+
+      for (const [eventName] of Object.entries(state.events)) {
+        machine.states[state.type.name]
+          .addEventListener(eventName, async (e) => await this.transition(e));
```

Comment:
Unnecessary await, surprised that eslint didn't catch that.

Reply:
Vestigial from a previous iteration, I'll do a pass to see what else might need cleanup.